ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/healthcheck/HealthCheckResponseTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.healthcheck;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.ksql.execution.json.PlanJsonMapper;
+import io.confluent.ksql.rest.entity.HealthCheckResponse;
+import io.confluent.ksql.rest.entity.HealthCheckResponseDetail;
+import java.io.IOException;
+import java.util.HashMap;
+import org.junit.Test;
+
+public class HealthCheckResponseTest {
+  private static final boolean IS_HEALTHY_BOOLEAN =  true;
+  private static final HealthCheckResponseDetail IS_HEALTHY_DETAIL =  new HealthCheckResponseDetail(IS_HEALTHY_BOOLEAN);
+
+  @Test
+  public void shouldDeserializeCorrectly() throws IOException {
```

Comment:
Can you add a check to see if it can deserialize correctly without the server state field in the json string?

Reply:
added